### ISSUE TITLE
Bias node?

### ISSUE THREAD

**Issue by Gerrit02** [makeyourownneuralnetwork/makeyourownneuralnetwork#31]
Hello,
probably one of my last questions I have about this book "Make your own neural network" !
In most lecture about neural networks (even about image or label recognition) they talk also about a bias.
So a bias node will be foreseen in input and hidden nodes.

1) How comes, in this example no bias has been used and still the performance is very high?

2) What I understand is a bias will never change (even the weights connected to that bias) during the learning cycle of the 
   neural network. Let's say you want make use of a bias (in the input layer and hidden layer) in your example, how will this influence the code and result, because as if 
   I understand now, within each itteration or learning cycle, here all the nodes and weights will be adapted?

**Comment by makeyourownneuralnetwork** [comment]
here is a blog post i did on [bias nodes](http://makeyourownneuralnetwork.blogspot.co.uk/2016/06/bias-nodes-in-neural-networks.html)

1. the example was chosen carefully to work well even without bias nodes - but using bias nodes is more general, see blog
2. the weights will change to the bias otherwise there is no learning

**Comment by Gerrit02** [comment]
Hello Rashid,
thank you for the beautifull explanation regarding the bias node in your blog.
However still some questions remains.

How do you assign the value 1 to the bias node in the input layer.
Within the MNIST data set you add a line with as label a value 1 without any characters behind?

And what if you use also a bias node in the hidden layer. It should also have the value 1.

For now you used 200 hidden nodes, but those nodes get their values from the input nodes (after forward propagation).

I guess for the hidden nodes you should do something simular as what you did for the input nodes:

self.onodes = output_nodes + 1. (So you get 201 hidden nodes).

The bias from the input is not connected via weights to the bias of the hidden node. (Agree on that).
But how do you make sure the bias of the hidden node is 1 and stays 1?
How to adapt the code for that, also in the backpropagation... ?

Tnx,
